docker_images/java/wrapper/src/main/java/glue/ModuleGlue.java

```diff
@@ -435,16 +436,16 @@ public void reset()
         }
 
         @Override
-        public DirectMethodResponse onMethodInvoked(String methodName, Object methodData, Object context)
+        public DirectMethodResponse onMethodInvoked(String methodName, DirectMethodPayload methodData, Object context)
         {
             System.out.printf("method %s called%n", methodName);
             if (methodName.equals(this._methodName))
             {
-                String methodDataString = new String((byte[]) methodData);
-                System.out.printf("methodData: %s%n", methodDataString);
+                Object methodDataObject = methodData.getPayload(Map.class);
```

Comment:
Our test takes payload of request in the type of `Map` only, so here we call `getPayload(Map.class)` to get the payload on the receiving device/module side.